pkg/ebpf/tracer_windows.go

```diff
@@ -179,6 +218,82 @@ func (t *Tracer) GetStats() (map[string]interface{}, error) {
 	}, nil
 }
 
+// NewDDAPIFilter returns a filter we can apply to the driver
+func newDDAPIFilter(direction C.uint64_t, ifaceIndex int, isIPV4 bool) (fd C.struct__filterDefinition) {
+	fd.filterVersion = C.DD_FILTER_SIGNATURE
+	fd.size = C.sizeof_struct__filterDefinition
+	fd.direction = direction
+
+	if isIPV4 {
+		fd.af = windows.AF_INET
+		fd.v4InterfaceIndex = (C.ulonglong)(ifaceIndex)
+	} else {
+		fd.af = windows.AF_INET6
+		fd.v6InterfaceIndex = (C.ulonglong)(ifaceIndex)
+	}
+
+	return fd
+}
+
+func (t *Tracer) setFilter(fd C.struct__filterDefinition) (err error) {
+	var id int64
+	err = windows.DeviceIoControl(t.driverHandle, C.DDFILTER_IOCTL_SET_FILTER, (*byte)(unsafe.Pointer(&fd)), uint32(unsafe.Sizeof(fd)), (*byte)(unsafe.Pointer(&id)), uint32(unsafe.Sizeof(id)), nil, nil)
+	if err != nil {
+		return log.Errorf("Failed to set filter: %s\n", err.Error())
+	}
+	return
+}
+
+// To capture all traffic for an interface, we create an inbound/outbound traffic filter
+// for both IPV4 and IPV6 traffic going to that interface
+func createFiltersForInterface(iface net.Interface) (filters []C.struct__filterDefinition) {
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, true))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_INBOUND, iface.Index, false))
+	filters = append(filters, newDDAPIFilter(C.DIRECTION_OUTBOUND, iface.Index, false))
+	return
+}
+
+func (t *Tracer) prepareDriverFilters() (err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		log.Errorf("Error getting interfaces: %s\n", err.Error())
+		return
+	}
+
+	for _, i := range ifaces {
+		log.Debugf("Setting filters for interface: %s [%+v]", i.Name, i)
+
+		for _, filter := range createFiltersForInterface(i) {
+			err = t.setFilter(filter)
+			if err != nil {
+				log.Warnf("Failed to set filter [%+v] on interface [%+v]: %s\n", filter, i, err.Error())
```

Comment:
I believe that you should return here so that the error is returned.

Reply:
I'll add the error return since it can always be removed later if necessary. 

For some further explanation: I did not return here because I was unsure whether or not we want failure to create a filter resulting in a failure of to set up the tracer/system probe. 

I'm not sure I have enough information at this point about what could go wrong setting a filter to say if we expect this to fail a lot. The only sample I have is my VM, which indicates it could return an error since it only ever failed when I had miscoded something.